server/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -78,23 +106,62 @@ void build(final Map<String, TokenizerFactory> tokenizers, final Map<String, Cha
         for (String tokenFilterName : tokenFilterNames) {
             TokenFilterFactory tokenFilter = tokenFilters.get(tokenFilterName);
             if (tokenFilter == null) {
-                throw new IllegalArgumentException("Custom Analyzer [" + name() + "] failed to find filter under name " +
+                throw new IllegalArgumentException("Custom Analyzer [" + name + "] failed to find filter under name " +
                     "[" + tokenFilterName + "]");
             }
             tokenFilter = tokenFilter.getChainAwareTokenFilterFactory(tokenizer, charFiltersList, tokenFilterList, tokenFilters::get);
             tokenFilterList.add(tokenFilter);
         }
 
-        this.customAnalyzer = new CustomAnalyzer(tokenizerName, tokenizer,
-                charFiltersList.toArray(new CharFilterFactory[charFiltersList.size()]),
-                tokenFilterList.toArray(new TokenFilterFactory[tokenFilterList.size()]),
-                positionIncrementGap,
-                offsetGap
+        return new AnalyzerComponents(tokenizerName, tokenizer,
+            charFiltersList.toArray(new CharFilterFactory[charFiltersList.size()]),
+            tokenFilterList.toArray(new TokenFilterFactory[tokenFilterList.size()])
         );
     }
 
     @Override
-    public CustomAnalyzer get() {
+    public Analyzer get() {
         return this.customAnalyzer;
     }
+
+    public static class AnalyzerComponents {
```

Comment:
Can we make this a top-level class, and move the `createComponents` factory method above into it?

Reply:
I think so, it used to be a hidden private class, but since its public anyway now there's no point of making it an inner class I think.